rules/numeric-separators-style.js

```diff
@@ -0,0 +1,151 @@
+	chunks[0] = chunks[0].replace(offset, '');
+	return chunks;
+}
+
+function parseNumber(string, isHex) {
+	const groups = isHex ? [string] : string.split(/(?=\.|e|E)/);
+	const wholePart = groups.find(grp => !grp.startsWith('.') && !grp.toLowerCase().startsWith('e'));
+	const decimalPart = groups.find(grp => grp.startsWith('.'));
+	const powerPartLowercase = groups.find(grp => grp.startsWith('e'));
+	const powerPartUppercase = groups.find(grp => grp.startsWith('E'));
+	const powerPart = powerPartLowercase || powerPartUppercase;
+	return {
+		wholePart,
+		decimalPart: decimalPart ? decimalPart.slice(1) : undefined,
+		exp: {
+			powerPart: powerPart ? powerPart.slice(1) : undefined,
+			e: powerPart ?
+				(powerPart === powerPartUppercase ? 'E' : 'e') :
+				undefined
+		}
+	};
+}
+
+function getFixedValue(notation, string, {minimumThreshold, preferedGroupLength}) {
+	string = removeLiteralNotation(notation, string);
+	const {wholePart, decimalPart, exp} = parseNumber(string, notation === literalNotations.hexadecimal);
+	const {powerPart, e} = exp;
+	const numberGroups = [];
+	const powerGroups = [];
+
+	if (wholePart) {
+		if (wholePart.length < minimumThreshold) {
+			numberGroups[0] = [wholePart];
+		} else if (wholePart.endsWith('n')) {
+			numberGroups[0] = getChunks(wholePart.replace('n', ''), preferedGroupLength, false);
+			numberGroups[0][numberGroups[0].length - 1] += 'n';
+		} else {
+			numberGroups[0] = getChunks(wholePart, preferedGroupLength, false);
+		}
+	}
+
+	if (decimalPart) {
+		numberGroups[1] = getChunks(decimalPart, preferedGroupLength, true);
+	}
+
+	if (powerPart) {
+		if (powerPart.replace('-', '').length < minimumThreshold) {
+			powerGroups.push(powerPart);
+		} else if (powerPart.startsWith('-')) {
+			powerGroups.push(...getChunks(powerPart.replace('-', ''), preferedGroupLength, false));
+			powerGroups[0] = '-' + powerGroups[0];
+		} else {
+			powerGroups.push(...getChunks(powerPart, preferedGroupLength, false));
+		}
+	}
+
+	const base = numberGroups.map(part => part.join('_')).join('.');
+
+	const power = powerGroups.length > 0 ? e + powerGroups.join('_') : '';
+	return addLiteralNotation(notation, base + power);
+}
+
+const create = context => ({
+	Literal: node => {
+		if ((typeof node.value === 'number' || node.bigint) && !isLegacyOctal(node)) {
+			const literalType = getNumberLiteralType(node);
+			const options = context.options[0] ? context.options[0][literalType] : defaultOptions[literalType];
+
+			const notation = literalNotations[literalType];
+			const unseparated = node.raw.replace(/_/g, '');
+			const fixed = getFixedValue(notation, unseparated, options);
+
+			if (node.raw !== fixed) {
+				context.report({
+					node,
+					message: 'Invalid group length in numeric value.',
+					fix: fixer => fixer.replaceText(node, fixed)
+				});
+			}
+		}
+	}
+});
+
+module.exports = {
+	create,
+	meta: {
+		type: 'suggestion',
+		docs: {
+			url: getDocumentationUrl(__filename)
+		},
+		fixable: 'code'
```

Comment:
`schema` required. Example: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/rules/better-regex.js#L102

Reply:
Thanks! I'm not really familiar with schemas tho. Can you tell me if it is valid? Default values apparently doesn't work as the integration test is failing because of an incorrect (inexistent) configuration.